app/src/main/java/org/oppia/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -443,40 +485,84 @@ class StatePlayerRecyclerViewAssembler private constructor(
 
   private fun maybeAddNavigationButtons(
     pendingItemList: MutableList<StateItemViewModel>,
+    rightPendingItemList: MutableList<StateItemViewModel>,
     hasPreviousState: Boolean,
     canContinueToNextState: Boolean,
     hasGeneralContinueButton: Boolean,
-    stateIsTerminal: Boolean
+    stateIsTerminal: Boolean,
+    shouldSplit: Boolean
   ) {
     val hasPreviousButton = playerFeatureSet.backwardNavigation && hasPreviousState
     when {
       hasGeneralContinueButton && playerFeatureSet.forwardNavigation -> {
-        pendingItemList += ContinueNavigationButtonViewModel(
-          hasPreviousButton,
-          previousNavigationButtonListener,
-          fragment as ContinueNavigationButtonListener
-        )
+        if (!shouldSplit) {
+          pendingItemList += ContinueNavigationButtonViewModel(
+            hasPreviousButton,
+            previousNavigationButtonListener,
+            fragment as ContinueNavigationButtonListener
+          )
+        } else {
+          rightPendingItemList += ContinueNavigationButtonViewModel(
+            false,
```

Comment:
Please add inline comments for all boolean values.

Reply:
That wasn't my code but I'll do it 😃

Reply:
Done